app/main.py

```diff
@@ -1,2 +1,123 @@
+
+from typing import List, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, elements: List[tuple] = None) -> None:
+        if elements is None:
+            elements = []
+        self._bucket_size = 8
+        self._bucket_resize = self._bucket_size * 2 / 3
+        while len(elements) > self._bucket_resize:
+            self._bucket_size *= 2
+            self._bucket_resize = self._bucket_size * 2 / 3
+        self._buckets = [[] for i in range(self._bucket_size)]
+        self._len_bucket = 0
+        if len(elements):
+            self._assign_buckets(elements)
+
+    def _assign_buckets(self, elements: List[tuple]) -> None:
+        for _key, _value in elements:
+            if isinstance(_key, list | dict | set):
+                raise KeyError(
+                    f'The key can`t be like this: "{_key}" is "{type(_key)}"'
+                )
+            hashed_value = hash(_key)
+            index = hashed_value % self._bucket_size
+
+            while self._buckets[index]:
+                if self._buckets[index][0] == _key:
+                    break
+                index = (index + 1) % self._bucket_size
+
+            self._buckets[index] = (_key, _value,)
+
+    def _get_buckets_full(self) -> list:
+        return [full_bucket for full_bucket in self._buckets if full_bucket]
+
+    def _resize(self, new_element: List[tuple]) -> None:
+        current_buckets = self._get_buckets_full()
+        current_buckets += new_element
+
+        while self._len_bucket > self._bucket_resize:
+            self._bucket_size *= 2
+            self._bucket_resize = self._bucket_size * 2 / 3
+
+        self._buckets = [[] for i in range(self._bucket_size)]
+        self._len_bucket = 0
+        self._assign_buckets(current_buckets)
+
+    def __setitem__(self, _key: Any, _value: Any) -> None:
+        self._len_bucket = len(self) + 1
+
+        if self._len_bucket > self._bucket_resize:
+            self._resize([(_key, _value)])
```

Comment:
it is not so good to do resize + add new element in one method
Must be Single Responsibility for method

Reply:
Maybe resize will just resize it without adding new element?